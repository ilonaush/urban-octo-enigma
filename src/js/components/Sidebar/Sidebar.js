import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {BrowserRouter, Link, withRouter} from 'react-router-dom';

import './Sidebar.styl';
import {MENU_ITEMS} from "../../constants";
import SidebarMenu from "../SIdebarMenu/SidebarMenu";

class Sidebar extends Component {
    render() {
        return (
            <div className='sidebar'>
                <Link to='/' className='heading'>Dashboard</Link>
                <hr/>
                <SidebarMenu />
            </div>
        );
    }
}

Sidebar.propTypes = {};

export default Sidebar;
