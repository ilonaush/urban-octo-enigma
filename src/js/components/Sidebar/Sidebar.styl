.sidebar
  height 100%
  width 260px
  z-index 1030
  background-color #f96332
  min-height 100%
  box-shadow 0 2px 22px 0 rgba(0,0,0,.2), 0 2px 30px 0 rgba(0,0,0,.35)
  a.heading
    color #fff
    text-decoration none
    text-align center
    padding 25px
    display block
