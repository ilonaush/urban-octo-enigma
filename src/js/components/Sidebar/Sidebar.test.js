import React from 'react';
import ReactDOM from 'react-dom';
import Sidebar from './Sidebar';
import {MemoryRouter} from "react-router";

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MemoryRouter><Sidebar /></MemoryRouter>, div);
});