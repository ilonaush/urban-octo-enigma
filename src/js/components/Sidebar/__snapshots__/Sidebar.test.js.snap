// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders without crashing 1`] = `
<div
  className="sidebar"
>
  <Link
    className="heading"
    replace={false}
    to="/"
  >
    <span
      alt=""
      className="icon"
      style={
        Object {
          "backgroundImage": "url('images/cat.svg",
        }
      }
    />
    Kitty Farm Dashboard
  </Link>
  <SidebarMenu />
</div>
`;
